src/files/compose/validateDappnodeCompose.ts

```diff
@@ -0,0 +1,273 @@
+import { Manifest } from "../manifest/types";
+import semver from "semver";
+import { Compose } from "./types";
+import { getIsCore } from "../../utils/getIsCore";
+import { params } from "./params";
+
+let aggregatedError = new AggregateError([]);
+
+/**
+ * Validates against custom dappnode docker compose specs.
+ * This function must be executed after the official docker schema
+ * @param param0
+ */
+export function validateDappnodeCompose({
+  composeUnsafe,
+  manifest
+}: {
+  composeUnsafe: Compose;
+  manifest: Manifest;
+}): void {
+  // clean the errors
+  aggregatedError = new AggregateError([]);
+  const isCore = getIsCore(manifest);
+  try {
+    // COMPOSE TOP LEVEL restrictions
+
+    validateComposeVersion(composeUnsafe);
+    validateComposeNetworks(composeUnsafe);
+
+    // COMPOSE SERVICE LEVEL restrictions
+
+    const cpServicesNames = Object.keys(composeUnsafe.services);
+
+    for (const cpServiceName of cpServicesNames) {
+      validateComposeServiceKeys(composeUnsafe, cpServiceName);
+      validateComposeServiceValues(composeUnsafe, isCore, cpServiceName);
+      validateComposeServiceNetworks(composeUnsafe, isCore, cpServiceName);
+
+      const cpServiceVolumes = composeUnsafe.services[cpServiceName].volumes;
+      if (cpServiceVolumes) {
+        for (const cpServiceVolume of cpServiceVolumes) {
+          validateComposeAndComposeServiceVolumes(
+            composeUnsafe,
+            isCore,
+            cpServiceName,
+            cpServiceVolume
+          );
+        }
+      }
+    }
+
+    if (aggregatedError.errors.length > 0) throw aggregatedError;
```

Comment:
It's weird to throw and immediately after catch, why not remove the try / catch?